### DISCUSSION TITLE
How to know SMS is received from mobile phone to server?

### DISCUSSION THREAD

**Discussion by lujamaharjan** [python-smpplib/python-smpplib#164] [Q&A] [answered]
Hi @eigenein, @podshumok 
I am implementing python smpplib to make custom SMS gateway.
I am successfully able to send SMS by using the code in readme.md file.

I go read client.py code in library. I see there is 
`def set_message_received_handler(self, func)` method
it is triggered after I sent SMS from script. but it is not triggered when is send SMS from mobile phone 
to that script running server.
could you please help to figure out how to confirm the SMS is received in script running server when I send SMS from 
phone?

**Comment by eigenein** [comment, marked as answer]
Hi @lujamaharjan,

Basically, any message sent by SMSC to your script ends up in the handler set by `set_message_received_handler`. Now, seeing that you successfully receive message delivery notifications, could you double check that your SMSC indeed sends out `deliver_sm` messages to you upon mobile-to-SMSC messages?

**Reply by lujamaharjan** [reply]
Hi, @eigenein 
Thank you for your giving time to help me. when we sent SMS from mobile phone to server we are successfully getting deliver_sm message.
Now we are getting sender's mobile number and sms content using `pdu.short_message` and `pdu.source_addr` inside 
`set_message_recived_handler`.
Once again thank you for replying and helping me.

with regards
lujamaharjan

**Reply by Woyram** [reply]
Hello @eigenein I really wish to understand your solution to the problem. I am still facing the issue. I sent a message from my phone to the server and I am still trying to figure out how I can get the message.

**Reply by lujamaharjan** [reply]
Hi @Woyram,

Basically `set_message_received_handler` function is executed when you sent message from your phone to server.

```python
client.set_message_received_handler(lambda pdu: sms_received_function(pdu))

def sms_received_function(pdu):
    print(pdu)
```
